modules/runner/src/main/java/org/apache/ignite/internal/configuration/DistributedConfigurationUpdater.java

```diff
@@ -31,37 +31,37 @@ public class DistributedConfigurationUpdater implements IgniteComponent {
 
     private static final IgniteLogger LOG = Loggers.forClass(DistributedConfigurationUpdater.class);
 
-    private final CompletableFuture<ConfigurationPresentation<String>> clusterCfgPresentation = new CompletableFuture<>();
+    private final ClusterManagementGroupManager cmgMgr;
 
-    public void setDistributedConfigurationPresentation(ConfigurationPresentation<String> presentation) {
-        clusterCfgPresentation.complete(presentation);
-    }
+    private final ConfigurationPresentation<String> presentation;
 
-    /**
-     * Applies changes to the cluster configuration when {@link DistributedConfigurationUpdater#clusterCfgPresentation}
-     * is complete.
-     *
-     * @param configurationToApply Cluster configuration that should be applied.
-     * @return Future that will be completed when cluster configuration is updated.
-     */
-    public CompletableFuture<Void> updateConfiguration(String configurationToApply) {
-        return clusterCfgPresentation.thenCompose(presentation -> presentation.update(configurationToApply))
-                .whenComplete((v, e) -> {
-                    if (e != null) {
-                        LOG.error("Unable to update cluster configuration", e);
-                    } else {
-                        LOG.info("Cluster configuration updated successfully");
-                    }
-                });
+    public DistributedConfigurationUpdater(ClusterManagementGroupManager cmgMgr, ConfigurationPresentation<String> presentation) {
+        this.cmgMgr = cmgMgr;
+        this.presentation = presentation;
     }
 
     @Override
     public void start() {
-
+        cmgMgr.clusterConfigurationToUpdate()
+                .thenApply(action -> {
```

Comment:
This should be `thenAccept`, not `thenApply` (as was written in my original comment), what's the point of creating a future in the `else` branch?